src/docs/binding-properties.md

```diff
@@ -0,0 +1,114 @@
+# Properties
+
+## Simple Properties on Swift Structs
+
+Properties in Swift structs have different calling conventions depending on whether the struct is frozen or not, and whether it's a getter or setter:
+
+- Getters:
+  - For frozen structs: Take the struct value lowered into registers
+  - For non-frozen structs: Take a pointer to the struct instance
+  
+- Setters:
+  - For both frozen and non-frozen structs: Always take a pointer to self and a value to set
+  - The pointer to self is required because setters need to modify the struct's memory in-place
+
+This means that when binding properties to C#, we need to handle these different cases appropriately:
+
+- For frozen struct getters, we can pass `SwiftSelf<T>`.
+- For non-frozen struct getters, we need to pass `SwiftSelf` pointer.
+- For all setters, we must always pass the new value and `SwiftSelf` pointer.
+
+Example of how this affects the binding:
+
+```swift
+@frozen public struct FrozenPoint {
+    public var x: Int
+}
+
+public struct NonFrozenPoint {
+    public var x: Int
+}
+```
+
+The generated C# binding needs to handle the different calling conventions:
+
+```csharp
+[StructLayout(LayoutKind.Sequential)]
+public struct FrozenPoint {
+    nint x;
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport("MySwiftModule", EntryPoint = "...")]
+    private static extern nint PInvoke_get_x(SwiftSelf<FrozenPoint> self);
+    
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport("MySwiftModule", EntryPoint = "...")]
+    private static extern void PInvoke_set_x(nint value, SwiftSelf self);
+
+    private unsafe nint Get_x()
+    {
+        var self = new SwiftSelf<FrozenPoint>(instance);
+        return PInvoke_get_x(self);
+    }
+
+    private unsafe void Set_x(nint value)
+    {
+        IntPtr payload;
+        try 
+        {
+            var metadata = ...;
+            payload = (SwiftHandle)NativeMemory.Alloc(returnMetadata.Size);
```

Comment:
I know this this is a general approach to Swift marshalling, however It isn't intuitive to allocate a new memory and later have `this` assignment:
```
this = SwiftMarshal.MarshalFromSwift<FrozenPoint>(payload);
```

Reply:
I think the key problem here is that we need to send the underlying memory of csharp struct into swift. And then Swift has to make some changes to that memory which later need to be visible on csharp side. Is this possible without making a copy?

Reply:
@jkotas, @AaronRobinsonMSFT  do you know?

I could not find a way - using fixed doesn't work as it doesn't allow fixing "this". And using GCHandle won't work since it needs to box the value -> creates a copy of the struct.

The runtime would need to be able to do 2 different things:
- If the C# property setter is called on a struct which is on a stack (so local variable, parameter, ...) then it's just "fixed" (no pinning required as the struct can't move)
- If the C# property setter is called on a struct which is on the heap (field on a class for example) then it would have to pin the "parent" class (GCHandle and all)

Reply:
Try this:
```csharp
struct MyStruct
{
    public void M()
    {
        fixed (MyStruct* p = &this)
        {
           ... do stuff with p ...
        }
    }
}
```

Reply:
I verified it works and replaced the relevant section